SpatialGDK/Source/SpatialGDK/Private/Tests/CrossServerRPCHandlerTest.cpp

```diff
@@ -0,0 +1,217 @@
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+
+	Builder = EntityComponentOpListBuilder();
+	Builder.AddEntityCommandRequest(TestEntityId, SuccessRequestId, CreateCrossServerCommandRequest());
+
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 2);
+	}
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_rpc_WHEN_unresolved_THEN_queue)
+{
+	AddExpectedError(QueueingCommand, EAutomationExpectedErrorFlags::Exact, 2);
+
+	ViewDelta Delta;
+	EntityView View;
+	ViewCoordinator Coordinator{ MakeUnique<MockConnectionHandler>(), nullptr };
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 1);
+	}
+
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_queued_rpc_WHEN_timeout_THEN_try_execute)
+{
+	AddExpectedError(QueueingCommand, EAutomationExpectedErrorFlags::Exact, 2);
+	AddExpectedError(ExecutingCommand, EAutomationExpectedErrorFlags::Exact);
+
+	ViewDelta Delta;
+	EntityView View;
+	ViewCoordinator Coordinator{ MakeUnique<MockConnectionHandler>(), nullptr };
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 1);
+	}
+
+	FPlatformProcess::Sleep(.5f);
```

Comment:
why .5 ? Can this be a defined variable somewhere which explains what it's for?

Reply:
I just need the process to sleep for a bit to make sure it gets processed. will create a field for it instead of using a magic number

Reply:
I'm possibly being silly, but what is it waiting for? I can't see anything on another thread here.